src/taipy/core/_scheduler/_scheduler.py

```diff
@@ -108,6 +130,24 @@ def _schedule_job_to_run_or_block(cls, job: Job):
                 cls.jobs_to_run.put(job)
             cls.__check_and_execute_jobs_if_development_mode()
 
+    @classmethod
+    def __wait_until_job_finished(cls, jobs: Union[List[Job], Job], timeout: Optional[int] = None):
+        def __check_if_timeout(start, timeout):
+            if timeout:
+                return (datetime.now() - start).seconds < timeout
+            return True
+
+        start = datetime.now()
+        jobs = jobs if isinstance(jobs, Iterable) else [jobs]
+
+        while __check_if_timeout(start, timeout):
+            try:
+                if all([job._is_finished() for job in jobs]):
+                    return
+                sleep(0.1)  # Limit CPU usage
```

Comment:
Here you are testing all jobs all the time.
What you could do is: 
- test jobs one by one in a loop
- pop out a finished job
- break the loop once a job is not finished (without passing to the next ones) 
- restart the loop with only the remaining jobs.

Maybe this can reduce the collision probability with the execution of the job itself.

You can also sleep one second instead of 0.1 second.

This is a proposal for a temporary workaround.

Reply:
I think 1 second is long enough to make the user feels conscious about the wait time, I'll lower it down to 0.5s, what do you think :D
Also I'm not sure I understand your 3rd point. Did you mean once a job is not finished and is timed out?

Reply:
On the third point, I mean we should continue, not break. Sorry. if a job is not finished, we don't need to check other jobs. We can go to the next iteration.